collector/fixtures/textfile/metrics_merge/a.prom

```diff
@@ -0,0 +1,5 @@
+# HELP events_total 
```

Comment:
Could we test cases with:

* No HELP string line at all
* 2x the same HELP string
* 2x different HELP strings

Reply:
Good idea! I've added these test-cases. I also changed the `ErrorHandling` in the `textfile` collector test to `ContinueOnError` because the reply we'd get for the `metrics_merge_different_help` test would look different "in the real life" (with the `HTTPErrorOnError`, we'd get an error message as the returned response body, whereas we get incomplete metrics when actually querying the `/metrics` endpoint).
Alternatively we can keep `ErrorHandling` as it was before and just have the test expect the error message instead of the incomplete metrics.

Reply:
Wonderful, thanks! Hmm yeah, good question. I think I would prefer returning and testing for the error. That way, we are testing closer to the real-world behavior, and I would have expected an error as well.

Reply:
I'm not sure I understand : real-world (`GET /metrics` with conflicting `HELP` on a given metric), we get an incomplete reply as currently represented in the test-case and an error is logged by the node exporter. So, for now, the client doesn't get an error and is blind to the issue. This is reflected in the proposed test-case with `ErrorHandling: ContinueOnError`. Do you think we should keep `ErrorHandling: HTTPErrorOnError` in the test-case instead?

Perhaps related: I imagine we could also just bail out when we have conflicting metrics HELPs and return only a `node_textfile_scrape_error 1` metric for the textfile collector, but that's a big change in behavior...

Reply:
Oh sorry. Indeed I wasn't aware / forgot that the Node Exporter also had set `ContinueOnError` set for its metrics handler. In that case, I guess as you have it now is actually fine.